lib/ddtrace/contrib/aws/instrumentation.rb

```diff
@@ -28,7 +28,7 @@ def annotate!(span, pin, context)
           span.service = pin.service
           span.span_type = pin.app_type
           span.name = context.safely(:resource)
-          span.resource = RESOURCE
+          span.resource = context.safely(:resource)
```

Comment:
This is a great change that improves how stats are computed in our system. I'd say that probably the `span.name` should be what we used to have for ` RESOURCE` (so `aws.command` or similar).

We can update this part so that your contribution will be merged in the 0.13 development branch. Thank you very much!

Reply:
Great thanks. 

I am still a bit confused how the `span.name` and `span.resource` works in the trace UI.

But based on this screen : 
![image](https://user-images.githubusercontent.com/6594266/38856111-cf47fd80-4225-11e8-8484-45000da7497f.png)

Wouldn't changing the `span.name` to `RESOURCE` change the trace view to show `aws.command` instead of the resource name (i.e `s3.head_object`) ?

Reply:
the `span.name ?` in your screenshot is actually the span name too. So, yes, you'd be correct.

The relationship between `name` and `resource` is roughly `group` to `subgroup`. In other words "a [resource] is a kind of [span name]". By setting the name to `aws.command`, you get the benefit of seeing `aws.command` be broken down into each of its constituent resources, although you won't see the resource name appear as the label for the visual element in the flamegraph.